src/components/LastBuildCard.test.tsx

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2020 RoadieHQ
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { render } from '@testing-library/react';
+import {
+  ApiRegistry,
+  ApiProvider,
+  UrlPatternDiscovery,
+  errorApiRef,
+} from '@backstage/core';
+import { rest } from 'msw';
+import { msw } from '@backstage/test-utils';
+import { setupServer } from 'msw/node';
+import { TravisCIApiClient, travisCIApiRef } from '../api';
+import { RecentTravisCIBuildsWidget } from '..';
+import { buildsResponseMock, entityMock } from '../mocks/mocks';
+import { ThemeProvider } from '@material-ui/core';
+import { lightTheme } from '@backstage/theme';
+import { MemoryRouter } from 'react-router-dom';
+
+const discoveryApi = UrlPatternDiscovery.compile('http://exampleapi.com');
+const errorApiMock = { post: jest.fn(), error$: jest.fn() };
+
+const apis = ApiRegistry.from([
+  [errorApiRef, errorApiMock],
+  [travisCIApiRef, new TravisCIApiClient({ discoveryApi })],
+]);
+describe('LastBuildCard', () => {
+  const worker = setupServer();
+  msw.setupDefaultHandlers(worker);
+
+  beforeEach(() => {
+    // jest.resetAllMocks();
+    worker.use(
+      rest.get(
+        'http://exampleapi.com/travisci/api/repo/RoadieHQ%2Fsample-service/builds?offset=0&limit=1',
+        (_, res, ctx) => res(ctx.json(buildsResponseMock))

```

Comment:
This response has limit 5 no? I guess this is here as the user can decide to fetch only the most recent N builds and this would have a corresponding response?

Reply:
For some reason when page loads, there are two requests going - one with limit 1and second with the limit set by the user. Could track it down and fix, but wanted only to add test in this pull request.

Reply:
Interesting. Yep we can look into it separately.